spec/helpers/spec-helper.coffee

```diff
@@ -32,7 +33,7 @@ window.convertTopLevelKeysToObjects = (data) ->
 
 window.respondWith = (server, url, options) ->
   if options.resultsFrom?
-    data = $.extend {}, options.data, results: resultsArray(options.resultsFrom, options.data[options.resultsFrom])
+    data = merge({}, options.data, { results: resultsArray(options.resultsFrom, options.data[options.resultsFrom]) })
```

Comment:
We need to use `lodash.merge` here because `resultsArray` sometime returns an empty array, and `Object.assign` just ignores the `results` key completely in those cases.

Reply:
Wait. So `Object.assign({}, { resultsArray: [] })` just returns `{}`?

Reply:
or does results array sometimes return `nil` and you need to handle that?

Reply:
Playing around with this in the console give me different behavior than when running the legacy tests. Let me revert this change and investigate this in a future PR.

Reply:
It looks like this is not an issue with empty values, but rather that `Object.assign` returning a new object instead of extending the existing object.

```
console.log("extend options data: ": $.extend({}, options.data)) 

LOG LOG: Object{extend options data: : Task{_onAssociatedCollectionChange: function(){ ... }, storageManager: _StorageManager{collections: ..., expectations: ...}, cid: 'c1413', attributes: Object{id: ..., project_id: ..., description: ..., title: ..., archived: ..., parent_id: ...}, _changing: false, _previousAttributes: Object{}, changed: Object{}, id: '1', _pending: false, constructor: function Task() { ... }, brainstemKey: 'tasks', paramRoot: 'task', urlRoot: '/api/tasks', _associationKeyBlacklist: function() { ... }, get: function(field, options) { ... }, className: function() { ... }, fetch: function(options) { ... }, destroy: function(options) { ... }, parse: function(resp, xhr) { ... }, updateStorageManager: function(resp) { ... }, dependenciesAreLoaded: function(loadOptions) { ... }, optionalFieldsAreLoaded: function(optionalFields) { ... }, associationsAreLoaded: function(associations) { ... }, setLoaded: function(state, options) { ... }, invalidateCache: function() { ... }, toServerJSON: function(method, options) { ... }, defaultJSONBlacklist: function() { ... }, createJSONBlacklist: function() { ... }, updateJSONBlacklist: function() { ... }, clone: function() { ... }, _parseResultsResponse: function(resp) { ... }, _linkCollection: function(collectionName, models, collectionOptions, field) { ... }, _collectionHasMany: function(associator) { ... }, _collectionBelongsTo: function(associator) { ... }, on: function(name, callback, context) { ... }, once: function(name, callback, context) { ... }, off: function(name, callback, context) { ... }, trigger: function(name) { ... }, stopListening: function(obj, name, callback) { ... }, listenTo: function(obj, name, callback) { ... }, listenToOnce: function(obj, name, callback) { ... }, bind: function(name, callback, context) { ... }, unbind: function(name, callback, context) { ... }, validationError: null, idAttribute: 'id', initialize: function(){ ... }, toJSON: function(options) { ... }, sync: function() { ... }, escape: function(attr) { ... }, has: function(attr) { ... }, set: function(key, val, options) { ... }, unset: function(attr, options) { ... }, clear: function(options) { ... }, hasChanged: function(attr) { ... }, changedAttributes: function(diff) { ... }, previous: function(attr) { ... }, previousAttributes: function() { ... }, save: function(key, val, options) { ... }, url: function() { ... }, isNew: function() { ... }, isValid: function(options) { ... }, _validate: function(attrs, options) { ... }, keys: function() { ... }, values: function() { ... }, pairs: function() { ... }, invert: function() { ... }, pick: function() { ... }, omit: function() { ... }}}
```

versus

```
console.log("assign options data: ": Object.assign({}, options.data))

LOG LOG: Object{assign options data: : Object{_onAssociatedCollectionChange: function(){ ... }, storageManager: _StorageManager{collections: ..., expectations: ...}, cid: 'c1413', attributes: Object{id: ..., project_id: ..., description: ..., title: ..., archived: ..., parent_id: ...}, _changing: false, _previousAttributes: Object{}, changed: Object{}, id: '1', _pending: false}}
```

Reply:
I did confirm that `lodash.merge` returns the same result and passes the singular test failure, so going to add that back into this PR.